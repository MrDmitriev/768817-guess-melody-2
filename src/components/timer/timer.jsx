import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {ActionCreator} from '../../reducer.js';

export class Timer extends Component {
  render() {
    const {timer} = this.props;
    return (
      <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
        <span className="timer__mins">{timer.min}</span>
        <span className="timer__dots">:</span>
        <span className="timer__secs">{timer.sec}</span>
      </div>
    );
  }

  componentDidMount() {
    const {tick, timer} = this.props;
    setTimeout(() => tick(timer), 1000);
  }

  componentDidUpdate() {
    const {tick, timer, showLoseTime} = this.props;
    if (timer.secLeft === 0) {
      showLoseTime();
    }
    return setTimeout(() => tick(timer), 1000);
  }
}

Timer.propTypes = {
  timer: PropTypes.shape({
    sec: PropTypes.number,
    min: PropTypes.number,
    secLeft: PropTypes.number,
  }),
  tick: PropTypes.func,
  showLoseTime: PropTypes.func,
};

export default connect(
    (state) => ({
      timer: state.timer,
    }),
    (dispatch) => ({
      tick: (timer) => dispatch(ActionCreator.tick(timer)),
      showLoseTime: () => dispatch(ActionCreator.showLoseTime()),
    })
)(Timer);
