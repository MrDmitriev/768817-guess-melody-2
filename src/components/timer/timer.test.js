import React from 'react';
import renderer from 'react-test-renderer';
import {Timer} from './timer.jsx';

it(`should match snapshot`, () => {
  const tick = jest.fn();
  const callbackHandler = jest.fn();
  const props = {
    timer: {
      sec: 10,
      min: 1,
      secLeft: 70,
    },
    tick,
    callbackHandler,
  };

  const wrapper = renderer.create(<Timer {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
