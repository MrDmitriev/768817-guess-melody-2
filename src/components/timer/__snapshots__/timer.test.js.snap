// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="timer__value"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    1
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    10
  </span>
</div>
`;
